include/itkTileMontage.hxx

```diff
@@ -245,7 +245,8 @@ TileMontage<TImageType, TCoordinate>::RegisterPair(TileIndexType fixed, TileInde
   m_PCM->SetObligatoryPadding(m_ObligatoryPadding);
   m_PCM->SetReleaseDataFlag(this->GetReleaseDataFlag());
   m_PCM->SetReleaseDataBeforeUpdateFlag(this->GetReleaseDataBeforeUpdateFlag());
-  m_PCMOptimizer->SetPixelDistanceTolerance(m_PositionTolerance);
+  typename PCMOptimizerType::SamplePeakOptimizerType::Pointer samplePeakOptimizer = m_PCMOptimizer->GetSamplePeakOptimizer();
```

Comment:
`GetSamplePeakOptimizer` -> `GetModifiableSamplePeakOptimizer` otherwise it does not compile with legacy off.

Reply:
Good catch, thanks -- added.